src/share/poudriere/jail.sh

```diff
@@ -125,6 +129,28 @@ delete_jail() {
 	cache_dir="${POUDRIERE_DATA}/cache/${JAILNAME}-*"
 	rm -rf ${POUDRIERED}/jails/${JAILNAME} ${cache_dir} || :
 	echo " done"
+	if [ "${CLEAN}" == "none" ]; then
+		return 0
+	fi
+	msg_n "Cleaning ${JAILNAME} data..."
+	case ${CLEAN} in
+		all)
+			find ${POUDRIERE_DATA} -name "${JAILNAME}" -type d | xargs rm -rf || :
+			;;
+		logs)
+			logs_dir="${POUDRIERE_DATA}/logs/bulk/${JAILNAME}-*"
+			rm -rf ${logs_dir} || :
```

Comment:
I think we should use `find | xargs` here (logs/packages/wrkdirs) like you did above. The list of logs/packages/wrkdirs may exceed the command line length limit.

You could simplify with something like this:

```
finddir=
depth=
case "${CLEAN}" in
all)      finddir="${POUDRIERE_DATA}" ;;
logs)     finddir="${POUDRIERE_DATA}/logs"; depth=1 ;;
packages) finddir="${POUDRIERE_DATA}/packages"; depth=1 ;;
wrkdirs)  finddir="${POUDRIERE_DATA}/wrkdirs"; depth=1 ;;
esac
if [ -n "${finddir}" ]; then
     find "${finddir}/" -name "${JAILNAME}-*" ${depth:+-maxdepth ${depth}} -print0 | xargs -0 rm -rf || :
fi
```

Note the use of `-print0|-0` here too.
Also note I changed logs to look in _/logs_ rather than _/logs/bulk_.  It's still possible we'll store logs outside of _bulk/_ eventually.